docs/todo-due-dates.md

```diff
@@ -0,0 +1,178 @@
+- Feature Name: `todo-due-dates`
+- Start Date: `2020-11-09`
+- RFC PR: (leave this empty)
+- Checkup Issue: (leave this empty)
+
+# Summary
+
+[summary]: #summary
+
+This RFC proposes enhancing the todo functionality with due days, allowing authors to add accountability and commitment to fix the todo items by a specified amount of time.
+
+# Motivation
+
+[motivation]: #motivation
+
+The lint todo functionality is intended to seamlessly integrate with existing linting tools, allowing the incremental fixing of linting errors on very large projects. However, one undesirable side effect of this is that developers may forget about fixing the todo items leading to the accumulation of technical debt.
+
+By adding a time threshold to the todo items, developers will now have a commitment to fix the linting problems because todo items that are past this threshold will have their severity increased accordingly.
+
+# How We Will Teach This
+
+[pedagogy]: #pedagogy
+
+- There will be documentation on how to configure when todo items are due to be fixed.
+- There will be a config and/or CLI argument to specify the number of days an item's severity changes, either to `warn` or to `error`.
+
+# Requirements
+
+[requirements]: #requirements
+
+- By default, todos are added with no due dates.
+- Authors can specify the number of days the todos become errors after its creation date.
+- Author can also specify the number of days the todos become warnings after its creation date. The error threshold must also be passed and must be greater than this threshold.
+- Configuration can be specified in a common file or overriden via CLI arguments.
+
+# Implementation
+
+[requirements]: #implementation
+
+## High Level Flow
+
+When an author runs a linter with the todo functionality, all errors are converted to todos with no due dates unless one is present in `package.json` or passed as a command line argument.
```

Comment:
```suggestion
When an author runs a linter with the todo functionality, all errors are converted to todos with no due date threshold unless one is present in `package.json` _or_ passed as a command line argument.
```

Reply:
Would we want to be able to specify a different due date for eslint vs. ember-template-lint? Does it make sense to have this threshold only configured in package.json?